plugins/package-managers/conan/src/main/kotlin/ConanV2Handler.kt

```diff
@@ -0,0 +1,267 @@
+                "info",
+                "-f",
+                "json",
+                "--out-file",
+                jsonFile.absolutePath,
+                *DUMMY_COMPILER_SETTINGS,
+                definitionFile.name
+            ).requireSuccess()
+        }
+
+        val pkgInfosV2 = parsePackageInfosV2(jsonFile).also { jsonFile.parentFile.safeDeleteRecursively() }
+
+        val packageList = removeProjectPackageV2(pkgInfosV2, definitionFile.name)
+        val packages = parsePackagesV2(packageList)
+        val projectInfo = findProjectPackageInfoV2(pkgInfosV2, definitionFile.name)
+        val projectPackage = generateProjectPackageV2(projectInfo, definitionFile, workingDir)
+
+        val dependenciesScope = Scope(
+            name = SCOPE_NAME_DEPENDENCIES,
+            dependencies = parseDependencyTreeV2(pkgInfosV2, projectInfo.requires, workingDir)
+        )
+        val devDependenciesScope = Scope(
+            name = SCOPE_NAME_DEV_DEPENDENCIES,
+            dependencies = parseDependencyTreeV2(pkgInfosV2, projectInfo.buildRequires, workingDir)
+        )
+
+        return HandlerResults(packages, projectPackage, dependenciesScope, devDependenciesScope)
+    }
+
+    override fun getConanDataFile(name: String, version: String, conanStorageDir: File, recipeFolder: String?): File? =
+        recipeFolder?.let { File(it).resolve("conandata.yml") }
+
+    override fun listRemotes(): ProcessCapture {
+        // List configured remotes in JSON format.
+        return ConanCommand.run("remote", "list", "-f", "json").requireSuccess()
+    }
+
+    override fun parseRemoteList(remoteList: String): List<Pair<String, String>> {
+        @Serializable
+        data class Remote(
+            val name: String,
+            val url: String,
+            val verifySsl: Boolean,
+            val enabled: Boolean
+        )
+
+        val remotes = JSON.decodeFromString<List<Remote>>(remoteList)
+        return remotes.filter { it.enabled }.map { it.name to it.url }
+    }
+
+    override fun runInspectFieldCommand(workingDir: File, pkgName: String, jsonFile: File) {
+        val path = if ("conanfile" in pkgName) {
+            pkgName
+        } else {
+            // For Conan 2, "conan inspect" need the path of the reference. See https://github.com/conan-io/conan/issues/12532.
+            ConanCommand.run(
+                workingDir,
+                "cache",
+                "path",
+                pkgName
+            ).requireSuccess().stdout.trim()
+        }
+
+        ConanCommand.run(
+            workingDir,
+            "inspect",
+            path,
+            "-f",
+            "json",
+            "--out-file",
+            jsonFile.absolutePath
+        ).requireSuccess()
+    }
+
+    /**
+     * Return the full list of packages, excluding the project level information.
+     */
+    private fun removeProjectPackageV2(pkgInfos: List<PackageInfoV2>, definitionFileName: String): List<PackageInfoV2> =
```

Comment:
Consider making all private functions that do not require class context top-level functions in the file (placed after the handler class).